### PR TITLE
test network nano bash updated with smartbft consensus

### PR DESCRIPTION
**Pull request by Param-S**
It requires the configtxgen & orderer smartbft changes from PR https://github.com/hyperledger/fabric/pull/3781

Signed-off-by: Parameswaran Selvam <parselva@in.ibm.com>

### CODE REVIEW

File: test-network-nano-bash/configtx.yaml [hunk 1]
```diff
@@ -385,3 +360,65 @@ Profiles:
...
+                RequestForwardTimeout: 2s
+                RequestComplainTimeout: 20s
+                RequestAutoRemoveTimeout: 3m0s
+                ViewChangeResendInterval: 5s
+                ViewChangeTimeout: 20s
+                LeaderHeartbeatTimeout: 1m0s
+                CollectTimeout: 1s
+                RequestBatchMaxBytes: 10485760
+                IncomingMessageBufferSize: 200
+                RequestPoolSize: 400
```
Comment (yacovm) [inline test-network-nano-bash/configtx.yaml:388]: Let's make it 100000

File: test-network-nano-bash/orderer1.sh [hunk 2]
```diff
@@ -17,8 +17,14 @@ export ORDERER_GENERAL_TLS_PRIVATEKEY="${PWD}"/crypto-config/ordererOrganization
...
 # following setting is not really needed at runtime since channel config has ca root certs, but we need to override the default in orderer.yaml
 export ORDERER_GENERAL_TLS_ROOTCAS="${PWD}"/crypto-config/ordererOrganizations/example.com/orderers/orderer.example.com/tls/ca.crt
-export ORDERER_GENERAL_BOOTSTRAPMETHOD=file
-export ORDERER_GENERAL_BOOTSTRAPFILE="${PWD}"/channel-artifacts/genesis.block
+if [ $# -gt 0 ] && [ "$1" = "smartbft" ]
+then
+    export ORDERER_GENERAL_BOOTSTRAPMETHOD=none
+    export ORDERER_CHANNELPARTICIPATION_ENABLED=true
+else
+    export ORDERER_GENERAL_BOOTSTRAPMETHOD=file
```
Comment (yacovm) [inline test-network-nano-bash/orderer1.sh:25]: i think we are deprecating the file method in v3.0 no? so let's make it always none.

File: test-network-nano-bash/network.sh [hunk 3]
```diff
@@ -82,19 +87,26 @@ networkStart() {
...
-
-    echo "Joining channel (peer2)..."
-    . ./peer2admin.sh && ./join_channel.sh
-
-    echo "Joining channel (peer3)..."
-    . ./peer3admin.sh && ./join_channel.sh
-
-    echo "Joining channel (peer4)..."
-    . ./peer4admin.sh && ./join_channel.sh
+    if [ "$ORDERER_TYPE" = "smartbft" ]; then
```
Comment (yacovm) [inline test-network-nano-bash/network.sh:90]: can we simplify and always use osnadmin and channel participation also for Raft?

File: test-network-nano-bash/orderer3.sh [hunk 4]
```diff
@@ -17,8 +17,14 @@ export ORDERER_GENERAL_TLS_PRIVATEKEY="${PWD}"/crypto-config/ordererOrganization
...
+then
+    export ORDERER_GENERAL_BOOTSTRAPMETHOD=none
+    export ORDERER_CHANNELPARTICIPATION_ENABLED=true
+else
+    export ORDERER_GENERAL_BOOTSTRAPMETHOD=file
+    export ORDERER_GENERAL_BOOTSTRAPFILE="${PWD}"/channel-artifacts/genesis.block
+fi
 export ORDERER_FILELEDGER_LOCATION="${PWD}"/data/orderer3
 export ORDERER_CONSENSUS_WALDIR="${PWD}"/data/orderer3/etcdraft/wal
 export ORDERER_CONSENSUS_SNAPDIR="${PWD}"/data/orderer3/etcdraft/wal
```
Comment (yacovm) [inline test-network-nano-bash/orderer3.sh:30]: we now have BFT also, so let's change WALDIR to be /data/orderer3/consensus/wal in all files. 
Also, why is WALDIR equal to SNAPDIR here?

**Comment by yacovm** [comment]
Looks OK-ish but CI fails and someone that is a a maintainer needs to review this

**Comment by denyeart** [comment]
@Mergifyio rebase

**Comment by denyeart** [comment]
@Param-S Please rebase to pick up the recent fabric-samples fixes and improvements. 
Were you planning to make test-network work with SmartBFT? Or just test-network-nano-bash?

**Comment by denyeart** [comment]
Completed in https://github.com/hyperledger/fabric-samples/pull/1038

**Pull request closed without merging**